packages/core/src/utils/Util.ts

```diff
@@ -1021,3 +1022,32 @@ export const getValueAsShapeList = <T>(val: Value<T>): Shape<T>[] => {
 };
 
 //#endregion
+
+//#region AST
+
+export type ErrorLoc = {
+  type: NodeType;
+  range: SourceRange;
+};
+
+export const toErrorLoc = (node: {
```

Comment:
Is this function only renaming and grouping some values? Looks like you had to do some work when feeding them to monaco anyway. Why is this step necessary?